android/src/toga_android/fonts.py

```diff
@@ -28,69 +28,72 @@ class Font:
     def __init__(self, interface):
         self.interface = interface
 
-    def typeface(self, *, default=Typeface.DEFAULT):
-        cache_key = (self.interface, default)
-        if typeface := _FONT_CACHE.get(cache_key):
-            return typeface
-
-        font_key = self.interface._registered_font_key(
-            self.interface.family,
-            weight=self.interface.weight,
-            style=self.interface.style,
-            variant=self.interface.variant,
-        )
+        # Check for a cached typeface.
         try:
-            font_path = _REGISTERED_FONT_CACHE[font_key]
+            self.native_typeface = _FONT_CACHE[self.interface]
         except KeyError:
-            # Not a pre-registered font
-            if self.interface.family not in SYSTEM_DEFAULT_FONTS:
-                print(
-                    f"Unknown font '{self.interface}'; "
-                    "using system font as a fallback"
+
+            # Check for a system font.
+            try:
+                self.native_typeface = {
+                    # None is left as a placeholder.
+                    # The default button font is not marked as bold, but it has a weight
+                    # of "medium" (500), which is in between "normal" (400), and "bold"
+                    # (600 or 700). To preserve this, we use the widget's original
+                    # typeface as a starting point rather than Typeface.DEFAULT.
```

Comment:
This could be more explicit about what`None` means:

```suggestion
                    # The default button font is not marked as bold, but it has a weight
                    # of "medium" (500), which is in between "normal" (400), and "bold"
                    # (600 or 700). To preserve this, we interpret SYSTEM as the 
                    # widget's original typeface.
```